src/connector.rs

```diff
@@ -45,12 +48,23 @@ where
     })
 }
 
+/// not implemented for MacOS
+pub fn https_connector<CA>(
```

Comment:
Similarly, I suspect you need `#[cfg(not(any(target_os = "macos", target_os = "windows", target_os = "ios")))]` to avoid the compilation error.